src/grappler/ngraph_optimizer.cc

```diff
@@ -0,0 +1,188 @@
+/*******************************************************************************
+ * Copyright 2017-2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include "ngraph_optimizer.h"
+
+#include "tensorflow/core/framework/attr_value.pb.h"
+#include "tensorflow/core/framework/node_def.pb.h"
+#include "tensorflow/core/grappler/clusters/cluster.h"
+#include "tensorflow/core/grappler/grappler_item.h"
+#include "tensorflow/core/grappler/op_types.h"
+#include "tensorflow/core/grappler/optimizers/custom_graph_optimizer_registry.h"
+#include "tensorflow/core/grappler/utils.h"
+#include "tensorflow/core/platform/protobuf.h"
+
+#include <iomanip>
+#if defined NGRAPH_DISTRIBUTED
+#include "ngraph/distributed.hpp"
+#endif
+
+#include <iostream>
+
+using namespace std;
+
+namespace tensorflow {
+namespace ngraph_bridge {
+
+Status NgraphOptimizer::Optimize(tensorflow::grappler::Cluster* cluster,
+                                 const tensorflow::grappler::GrapplerItem& item,
+                                 GraphDef* output) {
+  NGRAPH_VLOG(1) << "Here at NgraphOptimizer ";
+
+  // Convert the GraphDef to Graph
+  GraphConstructorOptions opts;
+  opts.allow_internal_ops = true;
+  opts.expect_device_spec = true;
+  Graph graph(OpRegistry::Global());
+  TF_RETURN_IF_ERROR(ConvertGraphDefToGraph(opts, item.graph, &graph));
+
+  // Get the nodes to be skipped
+  std::vector<string> skip_these_nodes;
```

Comment:
Can use a set instead of vector? Then we don't need to loop on the vector.

Reply:
Currently, used a vector because it was small list. Will be replacing it with a map since we will need the name as well as the id. This is being done in the separate PR so did not want make a lot of changes here.